### DISCUSSION TITLE
dTao stable vs dynamic staking mechanism

### DISCUSSION THREAD

**Discussion by vstam1** [opentensor/bits#6] [General Bittensor Discussion]
In the Rao branch [code](https://github.com/opentensor/subtensor/blob/rao/pallets/subtensor/src/staking/stake_ops.rs#L65-L73) , I came across two staking mechanisms: dynamic and stable. From what I understand, dynamic subnets use the new Uniswap v2 pools, while static pools involve a 1-to-1 swap between Tao and dTao.

How exactly does the dynamic/static subnet setup work? Which subnets will be static or dynamic? Can subnet owners choose between these options, or are only specific subnets designated as static? Additionally, does a static subnet provide a "risk-free" staking experience compared to the dynamic pools, or are there still risks involved? Would love to hear your thoughts!

**Comment by Nahum-qaelum** [comment]
Cam we work together I need a Bittensor-native engineer
